aws/sns/resources.py

```diff
@@ -21,20 +22,49 @@ def sns_topics():
     )
 
 
-def sns_subscription_attributes():
+def sns_parent_topic_exists(subscription):
     "https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sns.html#subscription"
-    return [
+    topicArn = ":".join(
+        re.split(":", subscription)[0:6]
+    )  # Naive method of divining topic from subscription
+    try:
         botocore_client.get(
             service_name="sns",
-            method_name="get_subscription_attributes",
+            method_name="get_topic_attributes",
             call_args=[],
-            call_kwargs={"SubscriptionArn": subscription["SubscriptionArn"]},
-            profiles=[subscription["__pytest_meta"]["profile"]],
-            regions=[subscription["__pytest_meta"]["region"]],
+            call_kwargs={"TopicArn": topicArn},
+        )
+        return True
+    except:
+        return False
+
+
+def sns_get_subscription_attrs(subscription):
+    "https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sns.html#subscription"
+    # This function catches the error thrown by the AWS API when attempting to query the attributes of
+    # a subscription that has no parent topic.
+    try:
+        attrs = (
+            botocore_client.get(
+                service_name="sns",
+                method_name="get_subscription_attributes",
+                call_args=[],
+                call_kwargs={"SubscriptionArn": subscription["SubscriptionArn"]},
+                profiles=[subscription["__pytest_meta"]["profile"]],
+                regions=[subscription["__pytest_meta"]["region"]],
```

Comment:
I still have an unfinished PR to get rid of these ugly parameters, but didn't have time to finish it up yet.

Reply:
Cool.